### ISSUE TITLE
Bug: std.json.JSONException@std/json.d(308): JSONValue is not an object

### ISSUE THREAD

**Issue by LordAlderaan** [abraunegg/onedrive#2886]
### Describe the bug

I get the error below when trying to sync local data to the OneDrive using _onedrive --sync --upload-only --no-remote-delete_. I also tried _onedrive --sync --resync --upload-only --no-remote-delete_ with the same result.

```
Reading configuration file: /home/*snip*/.config/onedrive/config
Configuration file successfully loaded
WARNING: D-Bus message bus daemon is not available; GUI notifications are disabled
Using IPv4 and IPv6 (if configured) for all network operations
Attempting to contact Microsoft OneDrive Login Service
Successfully reached Microsoft OneDrive Login Service
Configuring Global Azure AD Endpoints
WARNING: Application has been configured NOT to cleanup remote files that are deleted locally.
User Configured Rate Limit: 4000000
There are interrupted session uploads that need to be resumed ...
Attempting to perform a database vacuum to optimise database
Database vacuum is complete
std.json.JSONException@std/json.d(308): JSONValue is not an object
----------------
??:? pure @safe void std.exception.bailOut!(std.json.JSONException).bailOut(immutable(char)[], uint, scope const(char)[]) [0xe3b16f]
??:? pure @safe bool std.exception.enforce!(std.json.JSONException).enforce!(bool).enforce(bool, lazy const(char)[], immutable(char)[], uint) [0xd0668b]
??:? inout pure @property @trusted inout(std.json.JSONValue[immutable(char)[]]) std.json.JSONValue.objectNoRef() [0xf7af23d7]
??:? [0xc8baa3]
??:? [0xcdf223]
??:? [0xcdee4b]
??:? [0xd59823]
??:? void rt.dmain2._d_run_main2(char[][], uint, extern (C) int function(char[][])*).runAll() [0xf784970b]
??:? _d_run_main2 [0xf78494a3]
??:? _d_run_main [0xf78492d3]
??:? [0xd81fbf]
??:? __libc_start_main [0xf759273f]
```

### Operating System Details

```shell
Linux laraspberrypi 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux

No LSB modules are available.
Distributor ID:	Raspbian
Description:	Raspbian GNU/Linux 11 (bullseye)
Release:	11
Codename:	bullseye
```

### Client Installation Method

From 3rd Party Source (PPA, OpenSuSE Build Service etc)

### OneDrive Account Type

Personal

### What is your OneDrive Application Version

onedrive v2.5.2-1+np1+1.1

### What is your OneDrive Application Configuration

```shell
Reading configuration file: /home/*snip*/.config/onedrive/config
Configuration file successfully loaded
WARNING: D-Bus message bus daemon is not available; GUI notifications are disabled
Application version                          = onedrive v2.5.2-1+np1+1.1
Compiled with                                = LDC 2094
User Application Config path                 = /home/*snip*/.config/onedrive
System Application Config path               = /etc/onedrive
Applicable Application 'config' location     = /home/*snip*/.config/onedrive/config
Configuration file found in config location  = true - using 'config' file values to override application defaults
Applicable 'sync_list' location              = /home/*snip*/.config/onedrive/sync_list
Applicable 'items.sqlite3' location          = /home/*snip*/.config/onedrive/items.sqlite3
Config option 'drive_id'                     = 
Config option 'sync_dir'                     = /mnt/D/OneDrive
Config option 'enable_logging'               = true
Config option 'log_dir'                      = /var/log/onedrive/
Config option 'disable_notifications'        = false
Config option 'skip_dir'                     = 
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp|*.swp|*.partial
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 12
Config option 'monitor_fullscan_frequency'   = 12
Config option 'read_only_auth_scope'         = false
Config option 'dry_run'                      = false
Config option 'upload_only'                  = true
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'cleanup_local_files'          = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'disable_download_validation'  = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = true
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 777
Config option 'sync_file_permissions'        = 777
Config option 'space_reservation'            = 52428800
Config option 'application_id'               = d50ca740-c83f-4d1b-b616-12c519384f0c
Config option 'azure_ad_endpoint'            = 
Config option 'azure_tenant_id'              = 
Config option 'user_agent'                   = ISV|abraunegg|OneDrive Client for Linux/v2.5.2-1+np1+1.1
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 4000000
Config option 'operation_timeout'            = 3600
Config option 'dns_timeout'                  = 60
Config option 'connect_timeout'              = 10
Config option 'data_timeout'                 = 60
Config option 'ip_protocol_version'          = 0
Config option 'threads'                      = 8
Environment var 'XDG_RUNTIME_DIR'            = true
Environment var 'DBUS_SESSION_BUS_ADDRESS'   = true
Config option 'notify_file_actions'          = false

Selective sync 'sync_list' configured        = false

Config option 'sync_business_shared_items'   = false

Config option 'webhook_enabled'              = false
```

### What is your 'curl' version

```shell
curl 7.74.0 (arm-unknown-linux-gnueabihf) libcurl/7.74.0 OpenSSL/1.1.1w zlib/1.2.11 brotli/1.0.9 libidn2/2.3.0 libpsl/0.21.0 (+libidn2/2.3.0) libssh2/1.9.0 nghttp2/1.43.0 librtmp/2.3
Release-Date: 2020-12-09, security patched: 7.74.0-1.3+deb11u13
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp 
Features: alt-svc AsynchDNS brotli GSS-API HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL TLS-SRP UnixSockets
```

### Where is your 'sync_dir' located

Local

### What are all your system 'mount points'

```shell
/dev/mmcblk0p7 on / type ext4 (rw,noatime)
devtmpfs on /dev type devtmpfs (rw,relatime,size=3735680k,nr_inodes=933920,mode=755)
proc on /proc type proc (rw,relatime)
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,nodev,size=1599964k,nr_inodes=819200,mode=755)
tmpfs on /run/lock type tmpfs (rw,nosuid,nodev,noexec,relatime,size=5120k)
cgroup2 on /sys/fs/cgroup type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate,memory_recursiveprot)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
bpf on /sys/fs/bpf type bpf (rw,nosuid,nodev,noexec,relatime,mode=700)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=29,pgrp=1,timeout=0,minproto=5,maxproto=5,direct)
mqueue on /dev/mqueue type mqueue (rw,nosuid,nodev,noexec,relatime)
sunrpc on /run/rpc_pipefs type rpc_pipefs (rw,relatime)
debugfs on /sys/kernel/debug type debugfs (rw,nosuid,nodev,noexec,relatime)
tracefs on /sys/kernel/tracing type tracefs (rw,nosuid,nodev,noexec,relatime)
configfs on /sys/kernel/config type configfs (rw,nosuid,nodev,noexec,relatime)
fusectl on /sys/fs/fuse/connections type fusectl (rw,nosuid,nodev,noexec,relatime)
/dev/sda1 on /mnt/E type ext4 (rw,relatime)
/dev/mmcblk0p6 on /boot type vfat (rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=ascii,shortname=mixed,flush,errors=remount-ro)
/dev/sdb1 on /mnt/D type ext4 (rw,relatime)
tmpfs on /run/user/1000 type tmpfs (rw,nosuid,nodev,relatime,size=799980k,nr_inodes=199995,mode=700,uid=1000,gid=1000)
veracrypt on /tmp/.veracrypt_aux_mnt1 type fuse.veracrypt (rw,nosuid,nodev,relatime,user_id=0,group_id=0,allow_other)
/dev/mapper/veracrypt1 on /mnt/D/Documents type ext4 (rw,relatime)
```

### What are all your local file system partition types

```shell
NAME FSTYPE FSVER LABEL UUID                                 FSAVAIL FSUSE% MOUNTPOINT
loop0
│                                                                           
└─veracrypt1
     ext4   1.0         35bd88be-f16b-4ed7-8664-05ebf3fdfe78   45,8G    86% /mnt/D/Doc
sda                                                                         
└─sda1
     ext4   1.0         9355016d-5912-43dd-b2ae-06e8875a50b6  530,8G    37% /mnt/E
sdb                                                                         
└─sdb1
     ext4   1.0         39d0bc08-8630-45d1-8235-8c56d7eb6174    6,9T    47% /mnt/D
mmcblk0
│                                                                           
├─mmcblk0p1
│    vfat   FAT32 RECOVERY
│                       05C6-FFA8                                           
├─mmcblk0p2
│                                                                           
├─mmcblk0p5
│    ext4   1.0   SETTINGS
│                       7d6837ca-27ce-4e26-9ec0-9e5f1108f60f                
├─mmcblk0p6
│    vfat   FAT32 boot  01C0-5889                             201,8M    20% /boot
└─mmcblk0p7
     ext4   1.0   root  69372019-8bb5-40a2-8d70-4f0f99216c72    3,1G    68% /
```

### How do you use 'onedrive'

I use abraunegg's onedrive application daily to manually sync a copy of local (USB HDD) backups to the OneDrive cloud using the _onedrive --sync --upload-only --no-remote-delete_ command. 

The OneDrive is not used for anything else and not active on any other machine (windows or otherwise). When I need to remove old backup files from the OneDrive or do other maintenance I do this through a browser.

### Steps to reproduce the behaviour

Run _onedrive --sync --upload-only --no-remote-delete_ 
or _onedrive --sync --resync --upload-only --no-remote-delete_.

### Complete Verbose Log Output

```shell
Reading configuration file: /home/*snip*/.config/onedrive/config
Configuration file successfully loaded
Using 'user' configuration path for application config and state data: /home/*snip*/.config/onedrive
WARNING: D-Bus message bus daemon is not available; GUI notifications are disabled
Using the following path to store the runtime application log: /var/log/onedrive/
Using IPv4 and IPv6 (if configured) for all network operations
Attempting to contact Microsoft OneDrive Login Service
Successfully reached Microsoft OneDrive Login Service
Checking Application Version ...
Attempting to initialise the OneDrive API ...
Configuring Global Azure AD Endpoints
The OneDrive API was initialised successfully
Opening the item database ...
WARNING: Application has been configured NOT to cleanup remote files that are deleted locally.
User Configured Rate Limit: 4000000
Application Version:  onedrive v2.5.2-1+np1+1.1
Account Type:         personal
Default Drive ID:     *snip*
Default Root ID:      *snip*
Remaining Free Space: 509.60 GB (547178833510 bytes)
Sync Engine Initialised with new Onedrive API instance
All application operations will be performed in the configured local 'sync_dir' directory: /mnt/D/OneDrive
There are interrupted session uploads that need to be resumed ...
Attempting to perform a database vacuum to optimise database
Database vacuum is complete
std.json.JSONException@std/json.d(308): JSONValue is not an object
----------------
??:? pure @safe void std.exception.bailOut!(std.json.JSONException).bailOut(immutable(char)[], uint, scope const(char)[]) [0x9ee16f]
??:? pure @safe bool std.exception.enforce!(std.json.JSONException).enforce!(bool).enforce(bool, lazy const(char)[], immutable(char)[], uint) [0x8b968b]
??:? inout pure @property @trusted inout(std.json.JSONValue[immutable(char)[]]) std.json.JSONValue.objectNoRef() [0xf7a763d7]
??:? [0x83eaa3]
??:? [0x892223]
??:? [0x891e4b]
??:? [0x90c823]
??:? void rt.dmain2._d_run_main2(char[][], uint, extern (C) int function(char[][])*).runAll() [0xf77cd70b]
??:? _d_run_main2 [0xf77cd4a3]
??:? _d_run_main [0xf77cd2d3]
??:? [0x934fbf]
??:? __libc_start_main [0xf751673f]
```

**Comment by abraunegg** [comment]
@LordAlderaan 
Please follow the steps to prepare your system to build the client as detailed here: https://github.com/abraunegg/onedrive/blob/master/docs/install.md#building-from-source---high-level-requirements and compile following https://github.com/abraunegg/onedrive/blob/master/docs/install.md#building-using-dmd-reference-compiler

> ```
> ??:? pure @safe void std.exception.bailOut!(std.json.JSONException).bailOut(immutable(char)[], uint, scope const(char)[]) [0x9ee16f]
> ??:? pure @safe bool std.exception.enforce!(std.json.JSONException).enforce!(bool).enforce(bool, lazy const(char)[], immutable(char)[], uint) [0x8b968b]
> ??:? inout pure @property @trusted inout(std.json.JSONValue[immutable(char)[]]) std.json.JSONValue.objectNoRef() [0xf7a763d7]
> ??:? [0x83eaa3]
> ??:? [0x892223]
> ??:? [0x891e4b]
> ??:? [0x90c823]
> ??:? void rt.dmain2._d_run_main2(char[][], uint, extern (C) int function(char[][])*).runAll() [0xf77cd70b]
> ??:? _d_run_main2 [0xf77cd4a3]
> ??:? _d_run_main [0xf77cd2d3]
> ??:? [0x934fbf]
> ??:? __libc_start_main [0xf751673f]
>```

Unfortunately, because of your choice of distribution (Raspbian) the LDC compiler, despite having debugging enabled - these details get stripped because of how Raspbian packages the LDC compiler used to compile onedrive.

The only way to resolve this is to manually compile the client using the instructions above and when recompiling use the following command:
```
./configure --enable-debug; make clean; make;
```

Once this is done, the above part will be decoded correctly, this this issue can be investigated further.

**Comment by LordAlderaan** [comment]
When I try to install the DMD Compiler using _curl -fsS https://dlang.org/install.sh | bash -s dmd_ it gives me the following error:
_no DMD binaries available for aarch64_

So I installed the LDC Compiler but then when I tried to compile onedrive I got the following error:
```
(ldc-1.39.0)*snip*@*snip*:~/onedrive $ ./configure --enable-debug; make clean; make;
checking for a BSD-compatible install... /usr/bin/install -c
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for dmd... ldc2
checking version of D compiler... ./configure: line 2130: /home/*snip*/dlang/ldc-1.39.0/bin/ldc2: No such file or directory

configure: error: Compiler version insufficient, current compiler version , minimum version 1.18.0
make: *** No rule to make target 'clean'.  Stop.
make: *** No targets specified and no makefile found.  Stop.
```

Funny part is that the file /home/*snip*/dlang/ldc-1.39.0/bin/ldc2 does exist.

Am I doing something wrong?

**Comment by abraunegg** [comment]
@LordAlderaan 
> Am I doing something wrong?

No - you are not doing anything incorrect or wrong. A compounding problem is `Raspbian GNU/Linux 11 (bullseye)`.

I will have to fly partially blind in fixing the original issue for you. I will have a PR for you to test with your system + LDC 1.39.0 shortly.

**Comment by abraunegg** [comment]
@LordAlderaan 
Whilst I am working on a fix, can you please do the following:

* Provide via email a verbose debug log using the existing application version you have installed - generate this via following this process: https://github.com/abraunegg/onedrive/wiki/Generate-debug-log-for-support

**Comment by abraunegg** [comment]
@LordAlderaan 
OK .. so for *whatever* reason, installing LDC on Debian 11 aarch64 is *broken*.

If you installed this way:
```
curl -fsS https://dlang.org/install.sh | bash -s ldc
```

Please remove the installation / deactivate it - it is *broken* ... and probably needs to be flagged upstream as a packaging bug to the LDC developers.

So .. that being said, to prepare your environment *correctly* to build the PR, please ensure you do this:

```
sudo apt install build-essential
sudo apt install libcurl4-openssl-dev libsqlite3-dev pkg-config git curl ldc
```
If you utilise notifications, also add this:
```
sudo apt install libnotify-dev
```


To clone the PR to resolve your issue, you can use a script like the following:
```
#!/bin/bash

PR=2893

rm -rf ./onedrive-pr${PR}
git clone https://github.com/abraunegg/onedrive.git onedrive-pr${PR}
cd onedrive-pr${PR}
git fetch origin pull/${PR}/head:pr${PR}
git checkout pr${PR}

# Configure and Build
./configure --enable-debug --enable-notifications; make clean; make;
./onedrive --version
```

This script will create a local folder called `onedrive-pr2893` with the PR version.

To run the PR, you need to run the client from the PR build directory:
```
./onedrive <any other options needed>
```
To install the PR, you will need to perform sudo make install to install the PR version to your system.

When running the PR, your version should be: `onedrive v2.5.2-6-g5f089a5` or greater.

When encountering the issue, the client should now respond in the following manner (as tested on Debian 11 aarch64):
```
......
alex@debian-aarch64:~/onedrive-pr2893$ ./onedrive -s -v
No user or system config file found, using application defaults
Using 'user' configuration path for application config and state data: /home/alex/.config/onedrive
Using IPv4 and IPv6 (if configured) for all network operations
Attempting to contact Microsoft OneDrive Login Service
Successfully reached Microsoft OneDrive Login Service
Checking Application Version ...
Attempting to initialise the OneDrive API ...
Configuring Global Azure AD Endpoints
The OneDrive API was initialised successfully
Opening the item database ...
Application Version:  onedrive v2.5.2-6-g5f089a5
Account Type:         business
Default Drive ID:     b!bO8V7s9SSk6r7mWHpIjURotN33W1W2tEv3OXV_oFIdQimEdOHR-1So7CqeT1MfHA
Default Root ID:      01WIXGO5V6Y2GOVW7725BZO354PWSELRRZ
Remaining Free Space: 1012.31 GB (1086963438031 bytes)
Sync Engine Initialised with new Onedrive API instance
All application operations will be performed in the configured local 'sync_dir' directory: /home/alex/OneDrive
There are interrupted session uploads that need to be resumed ...
Attempting to restore file upload session using this session data file: /home/alex/.config/onedrive/session_upload.eq4Nc3L19lmaAOYt
Restore file upload session failed - cleaning up resumable session data file: /home/alex/.config/onedrive/session_upload.eq4Nc3L19lmaAOYt
Attempting to restore file upload session using this session data file: /home/alex/.config/onedrive/session_upload.fAxm7MzdNlKSUIiq
Restore file upload session failed - cleaning up resumable session data file: /home/alex/.config/onedrive/session_upload.fAxm7MzdNlKSUIiq
Attempting to restore file upload session using this session data file: /home/alex/.config/onedrive/session_upload.P6TznZtp9bF3Lkc2
Restore file upload session failed - cleaning up resumable session data file: /home/alex/.config/onedrive/session_upload.P6TznZtp9bF3Lkc2
Attempting to restore file upload session using this session data file: /home/alex/.config/onedrive/session_upload.31XNhozHsPfJb52w
Restore file upload session failed - cleaning up resumable session data file: /home/alex/.config/onedrive/session_upload.31XNhozHsPfJb52w
Attempting to restore file upload session using this session data file: /home/alex/.config/onedrive/session_upload.b7g5Ii68MkvwTGjr
The local file to upload does not exist locally anymore
Restore file upload session failed - cleaning up resumable session data file: /home/alex/.config/onedrive/session_upload.b7g5Ii68MkvwTGjr
Attempting to restore file upload session using this session data file: /home/alex/.config/onedrive/session_upload.cKJTriMDbYktUQB1
Restore file upload session failed - cleaning up resumable session data file: /home/alex/.config/onedrive/session_upload.cKJTriMDbYktUQB1
Fetching /delta response from the OneDrive API for Drive ID: b!bO8V7s9SSk6r7mWHpIjURotN33W1W2tEv3OXV_oFIdQimEdOHR-1So7CqeT1MfHA
Processing API Response Bundle: 1 - Quantity of 'changes|items' in this bundle to process: 0
Finished processing /delta JSON response from the OneDrive API
No changes or items that can be applied were discovered while processing the data received from Microsoft OneDrive
Performing a database consistency and integrity check on locally stored data
......
```

> [!NOTE]
> Building on 'aarch64' takes a while. You must also have enough free system memory at at least 1GB of system swap space to successfully compile.

**Comment by abraunegg** [comment]
@LordAlderaan 
Just as validation, on Debian 11, x86_64 CPU architecture, using 
```
curl -fsS https://dlang.org/install.sh | bash -s ldc
```

generates the following (it works without issue):
```
Run `~/dlang/install.sh --help` for usage information.

Downloading and unpacking https://github.com/ldc-developers/ldc/releases/download/v1.39.0/ldc2-1.39.0-linux-x86_64.tar.xz
######################################################################## 100.0%
Using dub 1.38.0 shipped with ldc-1.39.0

Run `source ~/dlang/ldc-1.39.0/activate` in your shell to use ldc-1.39.0.
This will setup PATH, LIBRARY_PATH, LD_LIBRARY_PATH, DMD, DC, and PS1.
Run `deactivate` later on to restore your environment.
alex@debian11:~$ git clone https://github.com/abraunegg/onedrive.git
Cloning into 'onedrive'...
remote: Enumerating objects: 8279, done.
remote: Counting objects: 100% (3273/3273), done.
remote: Compressing objects: 100% (617/617), done.
remote: Total 8279 (delta 2877), reused 2869 (delta 2649), pack-reused 5006 (from 1)
Receiving objects: 100% (8279/8279), 8.37 MiB | 10.82 MiB/s, done.
Resolving deltas: 100% (5866/5866), done.
alex@debian11:~$ cd onedrive
alex@debian11:~/onedrive$ source ~/dlang/ldc-1.39.0/activate
(ldc-1.39.0)alex@debian11:~/onedrive$ ./configure; make clean; make;
checking for a BSD-compatible install... /usr/bin/install -c
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for dmd... ldc2
checking version of D compiler... 1.39.0
checking for curl... yes
checking for sqlite... yes
configure: creating ./config.status
config.status: creating Makefile
config.status: creating contrib/pacman/PKGBUILD
config.status: creating contrib/spec/onedrive.spec
config.status: creating onedrive.1
config.status: creating contrib/systemd/onedrive.service
config.status: creating contrib/systemd/onedrive@.service
rm -f onedrive onedrive.o version
rm -rf autom4te.cache
rm -f config.log config.status
if [ -f .git/HEAD ] ; then \
        git describe --tags > version ; \
else \
        echo v2.5.2 > version ; \
fi
ldc2 -w -J. -O -L-lcurl -L-lsqlite3  -L-ldl src/main.d src/config.d src/log.d src/util.d src/qxor.d src/curlEngine.d src/onedrive.d src/webhook.d src/sync.d src/itemdb.d src/sqlite.d src/clientSideFiltering.d src/monitor.d src/arsd/cgi.d -ofonedrive
```

So the LDC issue you ran into on aarch64 ... is a compiler issue|bug, not an issue with this client and/or components used to configure and build.

**Comment by LordAlderaan** [comment]
It seems to be working fine with pr2893. Syncing files as we speak.

--Tjores

**Comment by abraunegg** [comment]
Thanks for the confirmation, will merge the PR into 'master'

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
